rendermodules/materialize/render_downsample_sections.py

```diff
@@ -0,0 +1,159 @@
+        [m['mipmapLevels'].pop(k, 0) for k in m['mipmapLevels'].keys() if k != level]
+        tilespecs.append(m)
+    tempjson = tempfile.NamedTemporaryFile(suffix=".json", mode='w', delete=True)
+    tempjson.close()
+    with open(tempjson.name, 'w') as f:
+        json.dump(tilespecs, f, indent=4)
+        f.close()
+    return tempjson.name
+
+
+class RenderSectionAtScale(RenderModule):
+    default_schema = RenderSectionAtScaleParameters
+    default_output_schema = RenderSectionAtScaleOutput
+
+    def run(self):
+
+        zvalues1 = self.render.run(
+            renderapi.stack.get_z_values_for_stack,
+            self.args['input_stack'])
+
+        if self.args['minZ'] == -1 or self.args['maxZ'] == -1:
+            self.args['minZ'] = min(zvalues1)
+            self.args['maxZ'] = max(zvalues1)
+        elif self.args['minZ'] > self.args['maxZ']:
+            raise RenderModuleException('Invalid Z range: {} > {}'.format(self.args['minZ'], self.args['maxZ']))
+        
+        zvalues1 = np.array(zvalues1)
+        zrange = range(int(self.args['minZ']), int(self.args['maxZ'])+1)
+        zvalues = list(set(zvalues1).intersection(set(zrange)))
+
+        if not zvalues:
+            raise RenderModuleException('No valid zvalues found in stack for given range {} - {}'.format(self.args['minZ'], self.args['maxZ']))
+
+        self.args['temp_stack'] = self.args['input_stack']
+
+        #intz = [int(z) for z in zvalues]
+
+        stack_has_mipmaps = check_stack_for_mipmaps(self.render, self.args['input_stack'], zvalues)
+
+        # check if there are mipmaps in this stack
+        if stack_has_mipmaps:
+            print("stack has mipmaps")
+            # clone the input stack to a temporary one with level 1 mipmap alone
+            temp_no_mipmap_stack = "{}_no_mml_t{}".format(self.args['input_stack'], time.strftime("%m%d%y_%H%M%S"))
+
+            # set level of mipmap to retain as 1 (level 1)
+            level = 1
+            mypartial = partial(create_tilespecs_without_mipmaps,
+                                self.render,
+                                self.args['input_stack'],
+                                level)
+
+            with renderapi.client.WithPool(self.args['pool_size']) as pool:
+                jsonfiles = pool.map(mypartial, zvalues)
+
+            # set temp_stack as temp_no_mipmap_stack
+            self.args['temp_stack'] = temp_no_mipmap_stack
+
+            # create stack - overwrites existing one
+            self.render.run(renderapi.stack.create_stack,
+                        self.args['temp_stack'])
+
+            # set stack state to LOADING
+            self.render.run(renderapi.stack.set_stack_state,
+                        self.args['temp_stack'],
+                        state='LOADING')
+
+            # import json files into temp_no_mipmap_stack
+            # this also sets the stack's state to COMPLETE
+            self.render.run(renderapi.client.import_jsonfiles_parallel,
```

Comment:
this could be slow for a small number of sections -- handling a long list of tilespecs rather than section-parallel jsons makes better use of the processing pool in the workflow's current use case.

Reply:
I did not quite get this one. Are you saying that each section should have its own json?. If so, it is creating one json for each z. If its not the case, can you please elaborate.

Reply:
I think the way that import_jsonfiles_parallel works is to spawn a pool in which a single worker recieves a single jsonfile from your input list -- so in the example of running this for a single section you are spinning up a pool of n processes, but only using one to actually send anything to the webservice.  import_tilespecs_parallel takes as input a list of tilespecs and a poolsize and runs however many processes are in your pool as ingest clients running evenly divided subsets of the input tilespecs.

Reply:
this is more of an optimization and definitely not blocking for this PR, it just means you don't have to track temporary jsons and speeds up processing.